dds/DCPS/GuidUtils.cpp

```diff
@@ -37,12 +37,27 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 
 namespace OpenDDS { namespace DCPS {
 
+OPENDDS_STRING
+to_string(const EntityId_t& entityId)
+{
+  OPENDDS_STRING ret;
+
+  std::size_t len = sizeof(entityId.entityKey);
+  for (std::size_t i = 0; i < len; ++i) {
+    ret += to_dds_string(unsigned(entityId.entityKey[i]), true);
```

Comment:
I appreciate that is is how the code was beforehand, but do we really need 16 calls to sprintf (complete with parsing of format string) (which is what to_dds_string is doing here) for converting a GUID (or any kind of byte sequence) to a hex string? I'd rather see something like:

char hex_char(unsigned char in)
{
  return (in < 10) ? ('0' + static_cast<uint16_t>(in)) : ('A' + static_cast<uint16_t>(in - 10u));
}

std::string hex_string(const char* buffer, size_t size)
{
  std::string result(size * 2, ' ');
  while (size != 0)
  {
    --size;

    result[size * 2] = hex_char((static_cast<unsigned char>(buffer[size]) & 0xF0u) >> 4u);
    result[size * 2 + 1] = hex_char(static_cast<unsigned char>(buffer[size]) & 0x0Fu);
  }
  return result;
}

(taken from one of my own projects)

Reply:
Not sure why it chose to format it that way, but you get the idea.